src/plugins/cosmos/components/modals/Staking/views/Stake.tsx

```diff
@@ -0,0 +1,246 @@
+  marketData
+}: StakeProps) => {
+  const {
+    clearErrors,
+    control,
+    formState: { errors, isValid },
+    handleSubmit,
+    setError,
+    setValue
+  } = useForm<StakingValues>({
+    mode: 'onChange',
+    defaultValues: {
+      [Field.FiatAmount]: '',
+      [Field.CryptoAmount]: ''
+    }
+  })
+
+  const values = useWatch({ control })
+  const cryptoError = get(errors, 'cryptoAmount.message', null)
+  const fiatError = get(errors, 'fiatAmount.message', null)
+  const fieldError = cryptoError ?? fiatError
+  const [percent, setPercent] = useState<number | null>(null)
+  const [activeField, setActiveField] = useState<InputType>(InputType.Crypto)
+
+  const amountRef = useRef<string | null>(null)
+
+  const bgColor = useColorModeValue('gray.50', 'gray.850')
+  const borderColor = useColorModeValue('gray.100', 'gray.750')
+
+  const { cosmosStaking } = useModal()
+
+  const onSubmit = (_: any) => {
+    // TODO: onContinue()
+  }
+
+  const cryptoYield = calculateYearlyYield(apr, values.cryptoAmount)
+  const fiatYield = bnOrZero(cryptoYield).times(marketData.price).toFixed(2)
+
+  const translate = useTranslate()
+
+  const handleCancel = () => {
+    cosmosStaking.close()
+  }
+
+  const handlePercentClick = (_percent: number) => {
+    const cryptoAmount = bnOrZero(cryptoAmountAvailable).times(_percent)
+    const fiat = bnOrZero(cryptoAmount).times(marketData.price)
+    if (activeField === InputType.Crypto) {
+      setValue(Field.FiatAmount, fiat.toString(), { shouldValidate: true })
+      setValue(Field.CryptoAmount, cryptoAmount.toString(), { shouldValidate: true })
+    } else {
+      setValue(Field.FiatAmount, fiat.toString(), { shouldValidate: true })
+      setValue(Field.CryptoAmount, cryptoAmount.toString(), { shouldValidate: true })
+    }
+    setPercent(_percent)
+  }
+
+  const handleInputChange = (value: string) => {
+    setPercent(null)
+    if (activeField === InputType.Crypto) {
+      const fiat = bnOrZero(value).times(marketData.price)
+      setValue(Field.FiatAmount, fiat.toString(), { shouldValidate: true })
+    } else {
+      const crypto = bnOrZero(value).div(marketData.price)
+      setValue(Field.CryptoAmount, crypto.toString(), { shouldValidate: true })
+    }
+  }
+
+  const handleInputToggle = () => {
+    const field = activeField === InputType.Crypto ? InputType.Fiat : InputType.Crypto
+    if (fieldError) {
+      // Toggles an existing error to the other field if present
+      clearErrors(fiatError ? Field.FiatAmount : Field.CryptoAmount)
+      setError(fiatError ? Field.CryptoAmount : Field.FiatAmount, {
+        message: 'common.insufficientFunds'
+      })
+    }
+    setActiveField(field)
+  }
+  // TODO: wire me up, parentheses are nice but let's get asset name from selectAssetNameById instead of this
+  const asset = (_ => ({
+    name: 'Osmosis',
+    symbol: 'OSMO',
+    caip19: assetId
+  }))(assetId) as Asset
+  return (
+    <AnimatePresence exitBeforeEnter initial={false}>
```

Comment:
I'm not sure this is gonna do anything, if you want a transition you can use `<SlideTransition>` that will animate on mount of the component, `<AnimatePresence>` will just animate transitions when mounting and un-mounting components that are children of it.

Reply:
Nice, thanks! Will also sneakily change this before merge on the PRs layered on top of this, if applicable.